build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchBuildCompletePlugin.java

```diff
@@ -0,0 +1,220 @@
+                spec.getParameters().getProjectDir().set(projectDir);
+                spec.getParameters().getFilteredFiles().addAll(getFlowProviders().getBuildWorkResult().map((result) -> {
+                    System.out.println("Build Finished Action: Collecting archive files...");
+                    List<File> files = new ArrayList<>();
+                    files.addAll(resolveProjectLogs(projectDir));
+                    if (files.isEmpty() == false) {
+                        files.addAll(resolveDaemonLogs(daemonsLogDir));
+                        files.addAll(getFileOperations().fileTree(gradleWorkersDir).getFiles());
+                        files.addAll(getFileOperations().fileTree(new File(projectDir, ".gradle/reaper/")).getFiles());
+                    }
+                    return files;
+                }));
+            });
+        }
+    }
+
+    private List<File> resolveProjectLogs(File projectDir) {
+        var projectDirFiles = getFileOperations().fileTree(projectDir);
+        projectDirFiles.include("**/*.hprof");
+        projectDirFiles.include("**/build/test-results/**/*.xml");
+        projectDirFiles.include("**/build/testclusters/**");
+        projectDirFiles.include("**/build/testrun/*/temp/**");
+        projectDirFiles.include("**/build/**/hs_err_pid*.log");
+        projectDirFiles.exclude("**/build/testclusters/**/data/**");
+        projectDirFiles.exclude("**/build/testclusters/**/distro/**");
+        projectDirFiles.exclude("**/build/testclusters/**/repo/**");
+        projectDirFiles.exclude("**/build/testclusters/**/extract/**");
+        projectDirFiles.exclude("**/build/testclusters/**/tmp/**");
+        projectDirFiles.exclude("**/build/testrun/*/temp/**/data/**");
+        projectDirFiles.exclude("**/build/testrun/*/temp/**/distro/**");
+        projectDirFiles.exclude("**/build/testrun/*/temp/**/repo/**");
+        projectDirFiles.exclude("**/build/testrun/*/temp/**/extract/**");
+        projectDirFiles.exclude("**/build/testrun/*/temp/**/tmp/**");
+        return projectDirFiles.getFiles().stream().filter(f -> Files.isRegularFile(f.toPath())).toList();
+    }
+
+    private List<File> resolveDaemonLogs(File daemonsLogDir) {
+        var gradleDaemonFileSet = getFileOperations().fileTree(daemonsLogDir);
+        gradleDaemonFileSet.include("**/daemon-" + ProcessHandle.current().pid() + "*.log");
+        return gradleDaemonFileSet.getFiles().stream().filter(f -> Files.isRegularFile(f.toPath())).toList();
+    }
+
+    public abstract static class BuildFinishedFlowAction implements FlowAction<BuildFinishedFlowAction.Parameters> {
+        interface Parameters extends FlowParameters {
+            @Input
+            Property<File> getUploadFile();
+
+            @Input
+            Property<File> getProjectDir();
+
+            @Input
+            ListProperty<File> getFilteredFiles();
+
+            @Input
+            Property<BuildScanExtension> getBuildScan();
```

Comment:
Hmm, I'm not sure we should consider this an input since we're just using it to register scan links. Does `@Internal` work here?

Reply:
this must be @Input. Everything else is not supported. Even though error messages in gradle are misleading

Reply:
Interesting. I'm surprised that's even serializable. This might bite us later.